### ISSUE TITLE
Cannot install features using DISM

### ISSUE THREAD

**Issue by iaingblack** [Microsoft/mssql-docker#33]
I think this is more the underlying windowsservercore container but I cannot see a github page for that. So posting here, sorry if incorrect. I am using an official Azure Server 2016 machine with Containers VM. Docker version 1.12.2-cs2-ws-beta and image microsoft/windowsservercore:4d83c32ad497. Trying to install IIS-NetFxExtensibility via DISM fails on Docker 1.14-dev too. This really prevents the images being useful. This image is the underlying images used for the SQL server images so it is affecting those also.

`dism /online /enable-feature /featurename:IIS-NetFxExtensibility /All`

`Error: 0x800f081f
The source files could not be found.
Use the "Source" option to specify the location of the files that are required to restore the feature. For more information on specifying a source location, see http://go.microsoft.com/fwlink/?LinkId=243077.`

I know it is not a problem with Windows 2016 as I can run this command from the Server 2016 docker host (not servercore admittedly but surely close enough) and it installs the feature as expected. Am I doing anything wrong?

**Comment by perrysk-msft** [comment]
[@iaingblack]

IIS-NetFxExtensibility feature is dependent on .net 3.5 and since it is not installed you are getting the 0x800f081f error.

Could you please pull latest .net image i.e docker pull microsoft/dotnet-framework:3.5 and use that image to install IIS-NetFxExtensibility.

**Comment by iaingblack** [comment]
Many thanks, I didn't realise this was the case. I did have a further problem however, the .NetFX3 files for that feature aren't on the image so I thought I would have to mount an ISO/wim to get it going, this was going to be a problem. But, I got a Server 2016 ISO and copied file; 

\sources\sxs\microsoft-windows-netfx3-ondemand-package.cab

to a folder on the local machine and pointed my DISM command to it. Place that file with your dockerfile and add the lines below and it copies it across and installs the feature. After that you are good to go. Thanks again Perry!

`COPY microsoft-windows-netfx3-ondemand-package.cab c:/sources/sxs/microsoft-windows-netfx3-ondemand-package.cab`
`RUN dism /online /enable-feature /featurename:NetFx3 /All /Source:C:\sources\sxs /quiet`
